docs/source/project_documentation/user_guide/model.md

```diff
@@ -0,0 +1,83 @@
+# Model Definitions in AaC
+
+The AaC `model` root key is used to represent the components of a system. These,
+for example, can be the services in a system following a Microservice
+Architecture (MSA) design.
+
+In its most basic form, a `model` must have a `name` that distinguishes it from
+other AaC definitions. At the same time, however, a `model` that defines nothing
+more than a `name` is of little use for anything more than representing external
+components or actors.
+
+It is recommended to include some amount of `behavior` definitions for each
+model as that allows users and the AaC package to understand the component as a
+part of the system. Some questions we can answer for `model`s with defined
+`behavior`s are:
+
+1. What does the component do?
+1. What inputs does it expect?
+1. What outputs can other components expect from this component?
+
+## Defining Systems Through Model Composition
+
+Whether your system is a monolith or a collection of microservices, the `model`
+root key is used to represent how components fit together as part of the system
+and what those components do as part of the system.
+
+To compose a system from individual `component`s, one uses the `components` key.
+
+```yaml
+model:
+  name: AlarmClock
+  description: A simple alarm clock
+  components:
+    - name: clock
+      type: Clock
+    - name: timer
+      type: ClockTimer
+    - name: alarm
+      type: ClockAlarm
+  behavior:
+    - ...
+```
```

Comment:
I've got the magic sauce for you.

```suggestion
```{eval-rst}
.. literalinclude:: ../../../../python/model/alarm_clock/alarm_clock.yaml
    :language: yaml
    :lines: 6-28
```
```

Reply:
I was between pushing for leveraging the referencing ability for Sphinx or really just do it in a go with that task later this iteration. Late night brain went with the simple path of leave it for now because of time crunch. But you had the info so that's easy too! Thanks!